src/ts/roam/navigation.ts

```diff
@@ -3,15 +3,29 @@ import {addDays, isValid, RoamDate} from '../date/common'
 import {RoamDb} from './roam-db'
 
 export const Navigation = {
-    baseUrl: () => {
+    baseUrl() {
         // https://roamresearch.com/#/app/roam-toolkit/page/03-24-2020
         const url = Browser.getActiveTabUrl()
         const parts = url.hash.split('/')
 
-        url.hash = parts.slice(0, 3).concat(['page']).join('/')
+        url.hash = parts.slice(0, 3).join('/')
         return url
     },
-
+    basePageUrl() {
+        const url = this.baseUrl()
+        url.hash = url.hash.concat('/page')
+        return url
+    },
+    getPageUrlByName(name: string) {
+        const page = RoamDb.getPageByName(name)
+        return this.getPageUrl(page[':block/uid'])
+    },
+    getPageUrl(uid?: string) {
+        if (!uid) {
+            return this.baseUrl().toString()
+        }
```

Comment:
what's use case for this?

Reply:
to get the daily notes page

Reply:
you can just call `baseUrl` explicitly..

Reply:
that does not return a string -- don't want to leak .toString() into app code. And we might need the baseUrl as is for some other purpose

Reply:
not sure what do you mean by `leak .toString`. you literally use this only to get baseUrl, and there is already function for that 😛  
I don't find it intuitive that that this function would silently "succeed" on falsey input =\

Reply:
I use this to convert eh baseUrl (which is the class URL) to a string. It's not the same thing. @Stvad Should I add a separate function for this? What should it be called? 

By leak `toString`, I mean exactly that. It needs to be in a function here.

Reply:
added a function `getDailyNotesUrl` for this @Stvad

Reply:
hrm. ok 😛